tvtk/tests/test_scene.py

```diff
@@ -0,0 +1,84 @@
+""" Tests for the garbage collection of TVTK Scene objects.
+"""
+# Author: Deepak Surti
+# Copyright (c) 2015, Enthought, Inc.
+# License: BSD Style.
+
+import unittest
+import weakref
+import gc
+import os
+
+from tvtk.pyface.tvtk_scene import TVTKScene
+
+class TestScene(unittest.TestCase):
+    def setUp(self):
+        self.object_collected = []
+
+    def object_collected_callback(self, reference):
+        self.object_collected.append(True)
+
+    def test_some_object_garbage_collected(self):
+        """ This is a sample test which demonstrates that after deleting
+            an object instance, it should not be necessary to force garbage
+            collection, which is not currently true for TVTKScene and Scene
+            qt, wx implementations.
+            See:test_tvtk_scene_garbage_collected
+                test_qt_scene_garbage_collected
+                test_wx_scene_garbage_collected
+        """
+        class A():
+            pass
+
+        a = A()
+        r = weakref.ref(a, self.object_collected_callback)
+        del a
+        self.assertEqual(self.object_collected, [True])
+        self.assertEqual(r(), None)
+
+    def test_tvtk_scene_garbage_collected(self):
+        """ Test the garbage collection of TVTScene object."""
+        scene = TVTKScene()
+        scene_weakref = weakref.ref(scene, self.object_collected_callback)
+        del scene
+        ################################################################
```

Comment:
Actually the `gc.collect` should happen at the beginning of the test method to make the test more robust. Let me work with them.